packages/core/src/transport/transport.ts

```diff
@@ -37,6 +41,14 @@ function addBatchTime(url: string) {
   )}`
 }
 
+let hasReportedBeaconError = false
+function reportBeaconError(e: unknown) {
```

Comment:
Sorry to be nitpicking, but IMO if we want to handle this behavior, we may want to have a more global solution: a monitor introduced in [your previous PR](https://github.com/DataDog/browser-sdk/pull/795) is also subject to this, since it can be triggered while reporting a monitored error. WDYT? Maybe having a monitor limit or avoiding sending similar errors multiple times could be better.

Reply:
No worries!
I am not sure to follow you in:
> a monitor introduced in your previous PR is also subject to this, since it can be triggered while reporting a monitored error.

Could you precise your thoughts here?

Reply:
Imagine we try to send a monitor log. The message is added to the monitoring `Batch`. Then, when it flushes itself via the [`flushPeriodically` callback](https://github.com/DataDog/browser-sdk/blob/master/packages/core/src/transport/transport.ts#L154-L157), if something goes wrong and trigger an unexpected error (`this.flush()` throws somehow), this error will be added to the monitoring `Batch`. And then we can loop: if the same thing goes wrong, we keep re-adding it. Maybe `this.flush()` fails before doing the request, so the impact is limited. But maybe it fails after doing the request, so we could potentially report issue indefinitly.

Reply:
as seen together IRL, that seems a pretty low level risk, let's go with this for now.